src/encoder.rs

```diff
@@ -326,18 +327,36 @@ impl<W: Write> Encoder<W> {
         self.w.write_le(0u8)?; // bg index
         self.w.write_le(0u8) // aspect ratio
     }
+
+    fn finish_writing(&mut self) -> io::Result<()> {
+        self.w.write_le(Block::Trailer as u8)
+    }
+
+    /// Returns writer instance used by this encoder
+    pub fn into_inner(mut self) -> io::Result<W> {
+        self.finish_writing()?;
+        // safety: forget prevents double drop
+        let w = unsafe { mem::ManuallyDrop::take(&mut self.w) };
+        mem::forget(self);
```

Comment:
Dangerous, leaks `buffer: Vec<u16>` and potentially more in the future.

Reply:
Fixed